src/AdvancedMH.jl

```diff
@@ -13,8 +13,12 @@ export MetropolisHastings, DensityModel, RWMH, StaticMH, StaticProposal, RandomW
 # Reexports
 export sample, MCMCThreads, MCMCDistributed
 
-# Abstract type for MH-style samplers.
-abstract type Metropolis <: AbstractMCMC.AbstractSampler end
+# Abstract type for MH-style samplers. Needs better name? 
+abstract type MHSampler <: AbstractMCMC.AbstractSampler end
+abstract type ProposalStyle end
+
+struct RandomWalk <: ProposalStyle end
+struct Static <: ProposalStyle end
```

Comment:
Oh, just noticed that probably we shouldn't reintroduce `RandomWalk` etc that were refactored in the master branch.
```suggestion
```